internal/backend/remote-state/s3/backend.go

```diff
@@ -374,48 +411,130 @@ func (b *Backend) Configure(obj cty.Value) tfdiags.Diagnostics {
-		DebugLogging:              logging.IsDebugOrHigher(),
-		IamEndpoint:               stringAttrDefaultEnvVar(obj, "iam_endpoint", "AWS_IAM_ENDPOINT"),
-		MaxRetries:                intAttrDefault(obj, "max_retries", 5),
-		Profile:                   stringAttr(obj, "profile"),
-		Region:                    stringAttr(obj, "region"),
-		SecretKey:                 stringAttr(obj, "secret_key"),
-		SkipCredsValidation:       boolAttr(obj, "skip_credentials_validation"),
-		SkipMetadataApiCheck:      boolAttr(obj, "skip_metadata_api_check"),
-		StsEndpoint:               stringAttrDefaultEnvVar(obj, "sts_endpoint", "AWS_STS_ENDPOINT"),
-		Token:                     stringAttr(obj, "token"),
-		UserAgentProducts: []*awsbase.UserAgentProduct{
+		AccessKey:              stringAttr(obj, "access_key"),
+		CallerDocumentationURL: "https://opentofu.org/docs/language/settings/backends/s3",
+		CallerName:             "S3 Backend",
+		SuppressDebugLog:       logging.IsDebugOrHigher(),
+		IamEndpoint:            stringAttrDefaultEnvVar(obj, "iam_endpoint", "AWS_IAM_ENDPOINT"),
+		MaxRetries:             intAttrDefault(obj, "max_retries", 5),
+		Profile:                stringAttr(obj, "profile"),
+		Region:                 stringAttr(obj, "region"),
+		SecretKey:              stringAttr(obj, "secret_key"),
+		SkipCredsValidation:    boolAttr(obj, "skip_credentials_validation"),
+		StsEndpoint:            stringAttrDefaultEnvVar(obj, "sts_endpoint", "AWS_STS_ENDPOINT"),
+		Token:                  stringAttr(obj, "token"),
+		UserAgent: awsbase.UserAgentProducts{
 			{Name: "APN", Version: "1.0"},
 			{Name: httpclient.DefaultApplicationName, Version: version.String()},
 		},
 	}
 
-	if policyARNSet := obj.GetAttr("assume_role_policy_arns"); !policyARNSet.IsNull() {
-		policyARNSet.ForEachElement(func(key, val cty.Value) (stop bool) {
+	if val, ok := boolAttrOk(obj, "use_legacy_workflow"); ok {
+		cfg.UseLegacyWorkflow = val
+	} else {
+		cfg.UseLegacyWorkflow = true
+	}
+
+	if val, ok := boolAttrOk(obj, "skip_metadata_api_check"); ok {
+		if val {
+			cfg.EC2MetadataServiceEnableState = imds.ClientDisabled
+		} else {
+			cfg.EC2MetadataServiceEnableState = imds.ClientEnabled
+		}
+	}
+
+	if val, ok := stringAttrOk(obj, "shared_credentials_file"); ok {
+		cfg.SharedCredentialsFiles = []string{val}
+	}
+
+	if val, ok := boolAttrOk(obj, "skip_metadata_api_check"); ok {
+		if val {
+			cfg.EC2MetadataServiceEnableState = imds.ClientDisabled
+		} else {
+			cfg.EC2MetadataServiceEnableState = imds.ClientEnabled
+		}
+	}
+
+	if value := obj.GetAttr("role_arn"); !value.IsNull() {
+		cfg.AssumeRole = configureAssumeRole(obj)
+	}
+
+	if val, ok := stringSliceAttrDefaultEnvVarOk(obj, "shared_credentials_files", "AWS_SHARED_CREDENTIALS_FILE"); ok {
+		cfg.SharedCredentialsFiles = val
+	}
+	if val, ok := stringSliceAttrDefaultEnvVarOk(obj, "shared_config_files", "AWS_SHARED_CONFIG_FILE"); ok {
+		cfg.SharedConfigFiles = val
+	}
+
+	ctx := context.TODO()
```

Comment:
TODO context? Intentional?

Reply:
Yes. Will create an issue to get rid of it in the future.